### ISSUE TITLE
Enhance uploads view

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#3646]
From https://github.com/nextcloud/android/issues/3111#issuecomment-428331916

> I'd rather argue that the uploads view should also get a three dot menu to offer similar actions than the file list view, while the menu should then only be present for successfully uploaded files. Would that work as an alternative way to handle this?

**Comment by tobiasKaminsky** [comment]
Duplicating all the actions is maybe too much, e.g. removing/adding favorites would then also have to add/remove favorite icon, etc.

**Comment by AndyScherzinger** [comment]
Alternatively we could just open the details view on click? 

That would basically be what @jancborchardt wanted to happen when clicking on the notification (which we now removed since there has only been a notification for the last uploaded file, which is why we all agreed to remove it completely)

**Comment by tobiasKaminsky** [comment]
> Alternatively we could just open the details view on click?

:+1: 

What would happen if we click "back" on detail?
- uploads view
- click on uploaded file
- see details
- press back
	- see uploads view again
	- see parent folder of the file

**Comment by AndyScherzinger** [comment]
In this case I'd say

>     * see uploads view again

--> it is a back navigation and back in this case means "uploads view"

**Comment by tobiasKaminsky** [comment]
Ok, but this will be a bit more work. But I guess that it is what user will expect.

**Comment by AndyScherzinger** [comment]
> But I guess that it is what user will expect.

That is what I thought. To get this right we probably just need to send a flag when launching the activity/fragment. Right now it is a fragment while we could discsuss to make it its own activity which would also solve the sharing issue with previawable files (atm a separate sharing implementation) becuase then it would just be a flag + check with a simple backstack navigation / back-pressed thing :)

**Comment by tobiasKaminsky** [comment]
Having details as an activity will probably cause also trouble, as activity start different (slide up from bottom), …

I think (without diving too deep into it), a fragment will work. But we then would have to move all those sharing/other common functions into an interface, so that e.g. FDA (for details in file list) and UploadActivity (for details in upload view) can handle it.
We can then also move common function into it (and do not use an Inferface, but an [abstract] class).

**Comment by AndyScherzinger** [comment]
Yeah, that might work but it would be nice if we then find a way to re-use it and also use it for the sharing action in the preview fragments...

**Comment by mario** [comment]
While I'll always advocate for one-activity app(s), if the way activity appears is your biggest trouble, there are certainly ways to change its enter and exit animations :)

**Comment by AndyScherzinger** [comment]
Imo the biggest issue is having a details fragment which we need to launch in 3 palces (atm used in one) which has a lot of dependencies to it's calling activity... (which is also why I thought it would be best to just give it an activity of its own since interfaces are nice bt having several activities implementing the exact same thing so they can use this fragment seems to be nonsense in my opinion.
